code/modules/food_and_drinks/machinery/coffeemaker.dm

```diff
@@ -408,3 +498,243 @@
 	. = ..()
 	atom_storage.max_slots = 4
 	atom_storage.set_holdable(list(/obj/item/coffee_cartridge))
+
+/*
+ * impressa coffee maker
+ * its supposed to be a premium line product, so its cargo-only, the board cant be therefore researched
+ */
+
+/obj/machinery/coffeemaker/impressa
+	name = "impressa coffeemaker"
+	desc = "An industry-grade Impressa Modello 5 Coffeemaker of the Piccionaia Home Appliances premium coffeemakers product line. Makes coffee from fresh dried whole beans."
+	icon = 'icons/obj/machines/coffeemaker.dmi'
+	icon_state = "coffeemaker_impressa"
+	circuit = /obj/item/circuitboard/machine/coffeemaker/impressa
+	initial_cartridge = null		//no cartridge, just coffee beans
+	brew_time = 15 SECONDS			//industrial grade, its faster than the regular one
+	density = TRUE
+	pass_flags = PASSTABLE
+	#define BEAN_CAPACITY 10	//amount of coffee beans that can fit inside
```

Comment:
This should be moved up to the very top of the file, and have an `#undef BEAN_CAPACITY` at the very bottom of the file

Reply:
Addressed in [1758d26](https://github.com/tgstation/tgstation/pull/70991/commits/1758d260bf38349a463fb0c4c94b19bde1396d9f)